base-theme/layouts/partials/webpack_url.html

```diff
@@ -1,10 +1,12 @@
 {{- $port := getenv "WEBPACK_PORT" | default 3001 -}}
+{{- $host := getenv "WEBPACK_HOST" | default "0.0.0.0" -}}
```

Comment:
I was thinking about this while we were working on the testing PR and I think it's a good idea to have an optional override for the host address in general. `0.0.0.0` only works as a bind argument, as it's not a real IP address and is just used as shorthand to mean "all IP addresses associated with this machine" in the context in which its being used. In this context, since `hugo server` is being run on your dev machine, it binds to `127.0.0.1` as well as your machine's local IP address likely assigned by your router. Since the request for the webpack assets is ultimately being made on the client device, `0.0.0.0` or `localhost` aren't going to work because the former isn't a real IP address and the latter refers to the client device and not the dev server.

I think defaulting to `localhost` here is fine, as most use cases are "I'm running the dev server on my machine and I need to see the site." In the case of testing on external devices on your local network, a dev can easily put their local IP address into `WEBPACK_HOST`. Since our bind arguments all still use `0.0.0.0`, that will work.